sema4ai/vscode-client/src/robo/actionPackage.ts

```diff
@@ -384,3 +409,250 @@ async function getTemplate(actionServerLocation: string) {
+            const actionServerLocation = await downloadOrGetActionServerLocation();
+            if (!actionServerLocation) {
+                return;
+            }
+
+            progress.report({ message: "Validating authentication" });
+            const verifyLoginOutput = await verifyLogin(actionServerLocation);
+            if (!verifyLoginOutput || !verifyLoginOutput.logged_in) {
+                window.showErrorMessage(
+                    "Action Server Not authenticated to Control Room, run: Sema4ai: Authenticate the Action Server to Control Room"
+                );
+                return;
+            }
+
+            progress.report({ message: "Finding organizations" });
+            const organizations = await listOrganizations(actionServerLocation);
+            if (organizations.length === 0) {
+                return;
+            }
+
+            progress.report({ message: "Choose organization" });
+            const organization = await chooseOrganization(organizations);
+            if (!organization) {
+                return;
+            }
+
+            const tempDir = path.join(os.tmpdir(), "vscode-extension", Date.now().toString());
+            try {
+                fs.mkdirSync(tempDir, { recursive: true });
+
+                progress.report({ message: "Building package" });
+                const packagePath = await buildPackage(actionServerLocation, workspaceDir.uri.fsPath, tempDir);
+                if (!packagePath) {
+                    return;
+                }
+
+                progress.report({ message: "Uploading package to Control Room" });
+                const actionPackage = await uploadActionPackage(actionServerLocation, packagePath, organization.id);
+                if (!actionPackage) {
+                    return;
+                }
+
+                progress.report({ message: "Waiting for package to be verified" });
+                const verified = await waitUntilPackageVerified(
+                    actionServerLocation,
+                    actionPackage,
+                    organization.id,
+                    progress
+                );
+                if (!verified) {
+                    return;
+                }
+
+                progress.report({ message: "Getting changelog input" });
+                const changelogInput = await askUserForChangelogInput();
+                if (!changelogInput) {
+                    return;
+                }
+
+                progress.report({ message: "Updating the package changelog to Control Room" });
+                const updated = await updateChangelog(
+                    actionServerLocation,
+                    actionPackage.id,
+                    organization.id,
+                    changelogInput
+                );
+                if (!updated) {
+                    return;
+                }
+            } catch (error) {
+                window.showErrorMessage(`Failed to publish action package: ${error.message}`);
```

Comment:
I guess that if you don't return here it'll end up showing both the error as well as the `package published successfully!`.